src/Plugins/SimplnxCore/src/SimplnxCore/Filters/Algorithms/ReshapeDataArray.hpp

```diff
@@ -0,0 +1,151 @@
+#pragma once
+
+#include "SimplnxCore/SimplnxCore_export.hpp"
+
+#include "simplnx/DataStructure/DataArray.hpp"
+#include "simplnx/DataStructure/StringArray.hpp"
+#include "simplnx/Filter/IFilter.hpp"
+#include "simplnx/Parameters/DynamicTableParameter.hpp"
+#include "simplnx/Utilities/DataArrayUtilities.hpp"
+#include "simplnx/Utilities/FilterUtilities.hpp"
+
+namespace nx::core
+{
+template <typename T>
+struct is_allowed_array_type : std::false_type
+{
+};
+
+template <typename T>
+struct is_allowed_array_type<DataArray<T>> : std::true_type
+{
+};
+
+template <>
+struct is_allowed_array_type<StringArray> : std::true_type
+{
+};
+
+template <typename ArrayType>
+typename std::enable_if<is_allowed_array_type<ArrayType>::value, Result<>>::type ReshapeArrayImpl(DataStructure& dataStructure, const DataPath& inputArrayPath, const DataPath& outputArrayPath,
+                                                                                                  const IFilter::MessageHandler& messageHandler)
```

Comment:
```suggestion
template <class Class>
concept AllowedType = is_allowed_array_type<Class>::value;

template <AllowedType ArrayType>
Result<> ReshapeArrayImpl(DataStructure& dataStructure, const DataPath& inputArrayPath, const DataPath& outputArrayPath, const IFilter::MessageHandler& messageHandler)
```

Simplify this to allow for singular return type. This lays groundwork for following changes ensuring lifetime of result value is not violated. I tested this locally and results were not being returned because the reference became invalidated after reassignment.